### PR TITLE
Add FTP support

### PR DESCRIPTION
**Pull request by chambm**
Including automatic FTP file decompression which required copy/pasting more code than I'd like to from an SO answer, but I didn't find any standard library way to do the same thing.

The existing implementation doesn't support FTP because the `requests` package doesn't support FTP. There's a requests_ftp package but it's pretty shoddy and doesn't do the automatic decompression that HTTP gets.

### CODE REVIEW

**Comment by chambm** [comment]
Seems like there are 'some' PEP8 violations. Is there an automatic formatting tool for that?

**Comment by chambm** [comment]
@martenson I see you migrated this originally. Do the changes look good?

File: data_managers/data_manager_fetch_gene_annotation/data_manager/data_manager.py [hunk 1]
```diff
@@ -16,36 +89,33 @@ def url_download(url):
...
+
+        # uncompress file if needed
+        cf = get_compressed_file(file_name)
+        if cf is not None:
+            uncompressed_file_name = os.path.splitext(file_name)[0]
+            with open(uncompressed_file_name, 'w+') as uncompressed_file:
+                shutil.copyfileobj(cf.accessor, uncompressed_file)
+            os.remove(file_name)
+            file_name = uncompressed_file_name
+    except IOError, e:
```
Comment (mvdbeek) [inline data_managers/data_manager_fetch_gene_annotation/data_manager/data_manager.py]: for py3 that should be `except IOError as e:`

File: data_managers/data_manager_fetch_gene_annotation/data_manager/data_manager.py [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
 import datetime
+import gzip
 import json
 import os
+import shutil
 import sys
 import uuid
+import zipfile
 
-import requests
```
Comment (mvdbeek) [inline data_managers/data_manager_fetch_gene_annotation/data_manager/data_manager.py]: Then we can also drop the requests requirement in the data manager xml.

File: data_managers/data_manager_fetch_gene_annotation/data_manager/data_manager.py [hunk 3]
```diff
@@ -1,12 +1,85 @@
+#!/usr/bin/python
```
Comment (mvdbeek) [inline data_managers/data_manager_fetch_gene_annotation/data_manager/data_manager.py]: Let' drop the shebang, we're not going to use it.

**Comment by bgruening** [comment]
ping @chambm :)

**Comment by chambm** [comment]
@bgruening I just realized the lint is failing in a completely different data manager. Why is that?

**Comment by nsoranzo** [comment]
@chambm I just fixed the error on the `master` branch, can you rebase or merge `master`?

**Comment by chambm** [comment]
Tried to rebase. How badly did I F it up? :1st_place_medal:

**Comment by nsoranzo** [comment]
> Tried to rebase. How badly did I F it up?

A bit :wink: 
@chambm I've rebased and pushed to your branch, please have a look if it seems correct.

**Comment by chambm** [comment]
Works for me!

**Comment by martenson** [comment]
@galaxybot test this

**Comment by bgruening** [comment]
Thanks!

**Pull request merged**